src/client/java/daos-java/find_java_home.sh

```diff
@@ -5,6 +5,7 @@ if [ -n  "$java_home_path" ]
 then
 	echo "$java_home_path"
 else
+	# shellcheck disable=SC2046
 	java_path="$(dirname $(readlink -f $(which java)))"
```

Comment:
I think we just want to properly quote here:

```suggestion
	# shellcheck disable=SC2046
	java_path="$(dirname "$(readlink -f "$(command -v java)")")"
```

(`command -v` to avoid [SC2230](https://www.shellcheck.net/wiki/SC2230) which I'm surprised we are not getting an error for).

Reply:
You're probably right but again, I'm not familiar with this code or how much testing it gets so I'd like to make as few changes as possible.

Reply:
I agree with @brianjmurrell fix proposal as I have experimented errors with missing quoting and the fix seems pretty safe.
I really think that it is an error or forget from the author.